apps/server/src/modules/room/repo/room-factory.ts

```diff
@@ -0,0 +1,18 @@
+import { Room, RoomProps } from '../domain';
+import { ObjectId } from 'bson';
+
+export type RoomParams = Omit<RoomProps, 'id' | 'updatedAt' | 'createdAt'> &
+	Partial<Pick<RoomProps, 'id' | 'updatedAt' | 'createdAt'>>;
+
+export class RoomDomainFactory {
+	public static build(params: RoomParams): Room {
+		const props = {
+			id: params.id || new ObjectId().toHexString(),
+			createdAt: params.createdAt || new Date(),
+			updatedAt: params.updatedAt || new Date(),
+			...params,
+		};
+
+		return new Room(props);
+	}
+}
```

Comment:
I think it might probably be a good idea to attach the entity in the factory in order to avoid having domain objects without a "real" entity reference. That would somewhat "standardize" the creation of domain objects. So we could rely always on domain objects that are connected to the UoW.
How exactly that should look like has still to be discussed.